src/main/scala/org/constellation/ConstellationNode.scala

```diff
@@ -26,15 +26,60 @@ import org.constellation.util.{APIClient, Metrics}
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
 
-object ConstellationNode extends StrictLogging {
+// scopt requires default args for all properties.
+// Make sure to check for null early -- don't propogate nulls anywhere else.
+case class CliConfig(externalIp: java.net.InetAddress = null,
+                     externalPort: Int = 0,
+                     debug: Boolean = false)
 
-  val ConstellationVersion = "1.0.10"
+object ConstellationNode extends StrictLogging {
 
+  //noinspection ScalaStyle
   def main(args: Array[String]): Unit = {
     logger.info("Main init")
+
+    import scopt.OParser
+    val builder = OParser.builder[CliConfig]
+    val parser1 = {
```

Comment:
ultimately I agree -- the different configuration sources should all play nicely together, and it would be nice to abstract over it better. I'll save it for later though.